mycroft/skills/core.py

```diff
@@ -290,10 +286,24 @@ def receive_handler(message):
                 logger.error(
                     "An error occurred while processing a request in " +
                     self.name, exc_info=True)
-
         if handler:
-            self.emitter.on(intent_parser.name, receive_handler)
-            self.events.append((intent_parser.name, receive_handler))
+            self.emitter.on(name, wrapper)
+            self.events.append((name, wrapper))
+
+    def register_intent(self, intent_parser, handler, need_self=False):
+        name = intent_parser.name
+        intent_parser.name = self.name + ':' + intent_parser.name
+        self.emitter.emit(Message("register_intent", intent_parser.__dict__))
+        self.registered_intents.append((name, intent_parser))
+        self.add_event(intent_parser.name, handler)
```

Comment:
This needs to pass on `need_self` to `add_event()` otherwise the decorated intent handlers will fail.